change some properities names in data classes like weather and clothi…

Hossam-Haroon opened the pull request:
change some properities names in data classes like weather and clothingItem for more readability

add testcase for getClothingRecommendation with one function that depends on getWeather function

Hossam-Haroon committed:
change some properities names in data classes like weather and clothingItem for more readability
add testcase for getClothingRecommendation with one function that depends on getWeather function

Mohamed-Elhanafy commented on PariStyle/src/main/kotlin/domain/usecase/GetClothingRecommendationUseCase.kt:
```diff
@@ -1,9 +1,91 @@
+        )
+    }
+
+    private fun getListOfClothesItemsWithDifferentConditions(): List<ClothingItem>{
```
we can rename this to availableClothingItems and change it to val
```suggestion
        private val availableClothingItems = listOf(
```

Hossam-Haroon replied:
maybe I should add UNKNOWN word to the enum class related to weatherCondition so I can use it in an object as empty clothingItem. what do you think ?

Mohamed-Elhanafy replied:
I think yes it would be good

Hossam-Haroon replied:
so the variable, what about this : clothingItemOptions instead of availableClothingItem ? , as they are options an possibilities I created so I feel it fits more

Mohamed-Elhanafy commented on PariStyle/src/main/kotlin/domain/usecase/GetClothingRecommendationUseCase.kt:
```diff
@@ -1,9 +1,91 @@
+            clothingItem.suitableConditions.contains(weatherCondition) &&
+                    checkTemperatureRange(temperature,clothingItem)
+        }
+        return suitableClothes.firstOrNull() ?: throw PariStyleException.SuitableClothesException(
```
you can return empty ClothingItem no need for exception

Hossam-Haroon replied:
maybe I should add UNKNOWN word to the enum class related to weatherCondition so I can use it in an object as empty clothingItem. what do you think ?

Mohamed-Elhanafy commented on PariStyle/src/main/kotlin/domain/usecase/GetClothingRecommendationUseCase.kt:
```diff
@@ -1,9 +1,91 @@
+
+    }
+
+    private fun checkTemperatureRange(temperature : Double, clothingItem: ClothingItem): Boolean{
```
this for more functional approach 
```suggestion
       private fun isWithinTemperatureRange(temperature: Double, clothingItem: ClothingItem) =
        temperature in clothingItem.suitableTemperatureRange.min..clothingItem.suitableTemperatureRange.max
```

Mohamed-Elhanafy reviewed (commented):
Nice Job

Hossam-Haroon committed:
turn a function that returns a list of options to a list instead,
change how a function looks by adding equals to return boolean as it is only one line of code,
add companion object for emptyClothingItem and replace it instead of the exception

Hossam-Haroon committed:
make the companion object private

Hossam-Haroon committed:
add a function to suggest clothes when the user provide a location that we get the weather from.
